src/hooks/useModal.tsx

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import useFocusTrap from "@charlietango/use-focus-trap";
+
+export function useModal(initialState = false) {
+  const [isOpen, setOpen] = React.useState(initialState);
+  const close = React.useCallback(() => setOpen(false), []);
+  const open = React.useCallback(() => setOpen(true), []);
+  const ref = React.useRef<HTMLElement>();
+  const focusTrapRef = useFocusTrap(isOpen);
+  const modalRef = React.useCallback(
```

Comment:
not sure i understand the focus trap stuff, you might want to add some comments explaining this.

Reply:
added a comment! It is to make sure you can only tab inside the modal when its open, otherwise you might focus stuff in the background and that can be annoying as a user